packages/flutter/lib/src/widgets/framework.dart

```diff
@@ -2640,10 +2641,13 @@ class BuildOwner {
           }
           return true;
         }());
-        if (!kReleaseMode && debugProfileBuildsEnabled) {
+        final bool isTimelineTracked = !kReleaseMode && _isProfileBuildsEnabledFor(element.widget);
+        if (isTimelineTracked) {
           Map<String, String> debugTimelineArguments = timelineArgumentsIndicatingLandmarkEvent;
           assert(() {
-            debugTimelineArguments = element.widget.toDiagnosticsNode().toTimelineArguments();
+            if (kDebugMode) {
```

Comment:
I know it is weird to check for `kDebugMode` here but there is an explicit test that wants to run profile builds with asserts enabled so this has to be done.  Removing this is technically a breaking change.

Reply:
i don't understand, this ran in profile-mode-with-asserts before, why can't it run in profile-mode-with-asserts now?

Reply:
We've explicitly put an exception to block it from working: https://github.com/flutter/flutter/blob/fb60ae53e553486559e1ae0f400c322825808651/packages/flutter/lib/src/foundation/diagnostics.dart#L1570:L1570

That exception states that you shouldn't do this for profile builds.  Jacob was also worried when I showed him that the DiagnosticsNode was being used.

The reason it worked in the past I believe is because there is an early out mechanism higher in toDiagnosticsNode for when you aren't tracking widget creation.

Reply:
Why not just delete the assert and replace it with `if (kDebugMode)`?

Reply:
> Why not just delete the assert and replace it with `if (kDebugMode)`?

I didn't do that because it wouldn't functionally be the same since someone can run a debug build with the asserts removed.

Reply:
I think we've really tried to discourage customers from running profile with asserts or debug without. It also seems strange to guard this with "debug + asserts" rather than just debug or just asserts. But it'd be fine to try to tackle this problem more wholistically rather than hanging up this PR on it.